### DISCUSSION TITLE
Can I stop the training once i reached a desirable outcome?

### DISCUSSION THREAD

**Discussion by HarounAbdelsamad** [explosion/spaCy#13172] [Help: Coding & Implementations]
I am training an en_core_web_trf pre model for the last 15  hours and the process still did not end? Can I end it manually using ctrl + c or will this affect the model. I used the python -m spacy train config.cfg --output ./output from the beginning so it is supposed to be saving each model.
Here is the process:
"""
============================= Training pipeline =============================
ℹ Pipeline: ['transformer', 'ner']
ℹ Initial learn rate: 0.0
E    #       LOSS TRANS...  LOSS NER  ENTS_F  ENTS_P  ENTS_R  SCORE
---  ------  -------------  --------  ------  ------  ------  ------
  0       0        1067.52   1674.37    0.00    0.00    0.22    0.00
  0     200       85731.14  33170.92   71.12   75.67   67.09    0.71
  1     400       42521.32   1832.88   86.16   90.75   82.01    0.86
  2     600      108414.21   2911.67   89.98   90.49   89.48    0.90
  3     800       11231.89    896.99   86.07   80.68   92.24    0.86
  3    1000         623.92    605.81   87.16   89.03   85.37    0.87
  4    1200        1223.48    676.04   90.20   86.61   94.10    0.90
  5    1400         818.14    535.47   89.96   87.73   92.31    0.90
  6    1600         171.46    298.32   91.07   95.42   87.09    0.91
  6    1800         157.00    327.35   93.06   91.19   95.00    0.93
  7    2000        1104.12    333.76   93.00   92.29   93.73    0.93
  8    2200         120.79    229.26   93.76   95.58   92.01    0.94
  9    2400         399.62    244.94   93.02   91.12   95.00    0.93
 10    2600          74.70    176.95   94.35   93.93   94.78    0.94
 10    2800          73.17    154.58   94.53   94.35   94.70    0.95
 11    3000          72.04    130.66   93.80   93.28   94.33    0.94
 12    3200          77.78    164.30   93.35   91.48   95.30    0.93
 13    3400         157.12    125.21   93.83   93.42   94.25    0.94
"""
And here is my config file
"""
[paths]
train = "train.spacy"
dev = "dev.spacy"
vectors = null
init_tok2vec = null

[system]
gpu_allocator = "pytorch"
seed = 0

[nlp]
lang = "en"
pipeline = ["transformer","ner"]
batch_size = 128
disabled = []
before_creation = null
after_creation = null
after_pipeline_creation = null
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}
vectors = {"@vectors":"spacy.Vectors.v1"}

[components]

[components.ner]
factory = "ner"
incorrect_spans_key = null
moves = null
scorer = {"@scorers":"spacy.ner_scorer.v1"}
update_with_oracle_cut_size = 100

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "ner"
extra_state_tokens = false
hidden_width = 64
maxout_pieces = 2
use_upper = false
nO = null

[components.ner.model.tok2vec]
@architectures = "spacy-transformers.TransformerListener.v1"
grad_factor = 1.0
pooling = {"@layers":"reduce_mean.v1"}
upstream = "*"

[components.transformer]
factory = "transformer"
max_batch_items = 4096
set_extra_annotations = {"@annotation_setters":"spacy-transformers.null_annotation_setter.v1"}

[components.transformer.model]
@architectures = "spacy-transformers.TransformerModel.v3"
name = "roberta-base"
mixed_precision = false

[components.transformer.model.get_spans]
@span_getters = "spacy-transformers.strided_spans.v1"
window = 128
stride = 96

[components.transformer.model.grad_scaler_config]

[components.transformer.model.tokenizer_config]
use_fast = true

[components.transformer.model.transformer_config]

[corpora]

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[training]
output_directory = "roberta7030"
accumulate_gradient = 3
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
seed = ${system.seed}
gpu_allocator = ${system.gpu_allocator}
dropout = 0.1
patience = 1600
max_epochs = 0
max_steps = 20000
eval_frequency = 200
frozen_components = []
annotating_components = []
before_to_disk = null
before_update = null

[training.batcher]
@batchers = "spacy.batch_by_padded.v1"
discard_oversize = true
size = 2000
buffer = 256
get_length = null

[training.logger]
@loggers = "spacy.ConsoleLogger.v1"
progress_bar = false

[training.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = false
eps = 0.00000001

[training.optimizer.learn_rate]
@schedules = "warmup_linear.v1"
warmup_steps = 250
total_steps = 20000
initial_rate = 0.00005

[training.score_weights]
ents_f = 1.0
ents_p = 0.0
ents_r = 0.0
ents_per_type = null

[pretraining]

[initialize]
vectors = ${paths.vectors}
init_tok2vec = ${paths.init_tok2vec}
vocab_data = null
lookups = null
before_init = null
after_init = null

[initialize.components]

[initialize.tokenizer]
"""
